darwin/Platform.c

```diff
@@ -168,6 +172,18 @@ bool Platform_init(void) {
    return true;
 }
 
+// Converts ticks in the Mach "timebase" to nanoseconds.
+// See `mach_timebase_info`, as used to define the `Platform_nanosecondsPerMachTick` constant.
+uint64_t Platform_machTicksToNanoseconds(uint64_t mach_ticks) {
+   uint64_t ticks_quot = mach_ticks / Platform_nanosecondsPerMachTickDenom;
+   uint64_t ticks_rem  = mach_ticks - ticks_quot * Platform_nanosecondsPerMachTickDenom;
```

Comment:
Maybe keep it simple? (A compiler should recognize both as equivalent, but if it doesn't, my testing long ago showed that the `%` operator produces smaller code in `-Os` optimization.)

```suggestion
   uint64_t ticks_rem  = mach_ticks % Platform_nanosecondsPerMachTickDenom;
```

Reply:
Actually there usually is a divmod intrinsic in the compiler that should kick in if both a division and a modulo are done on the same arguments near each other. That said, using `__uint128` is non-standard and except if there's a **really** good reason for it, we should avoid it.